orbs/helm/circle-wait-job

```diff
@@ -94,19 +94,44 @@ uri = URI::HTTPS.build(host: "circleci.com",
                                                    "shallow" => true,
                                                    "filter" => "running"}))
 
-
+# CircleCI doesn't have built in way to avoid running to job at the same time on the same branch.
+# It means we have handle it manually.
+# Our main goal here is to avoid two helm deployement at the same time. It may impact badly the stability of the plateform.
+#
+# The easiest way to achieve is to add a task at the beginning of the job which wait if another build is running the same job.
+# It works fine until a deadlock, where two jobs are waiting each other.
+#
+# The issue come from the difficulty to known if the other job is already deploying our just waiting.
+# In case of concurrency, we can favor the build with the older committer_date, but we have no garanty that the other one is not deploying.
+# Indeed, a workflow can be runned faster than another one leading to a newest committer_date being deployed before the older.
+# Then, the use of committer_date avoid deadlock but may lead to concurrent deployment.
+#
+# Because we cannot determine if concurrent builds are already deploying or not, the most safe way to avoid both, concurrent deployment and deadlock, is to follow a first in first out logic.
+# The first build may not be aware of the second build and may have started to deploy.
+# At the opposite, the second build is always aware of the first build and can wait for its completion before deploying.
+# There is no risk of concurrency neither deadlock.
+#
+# We can use the build_num to easily know the first build. However, the builnumber is set at the queue time and not at the start time.
+# It means it correspond to the order of queue and not start.
+# It is possible for a build to start before another build that has been queued first.
+# I would be more correct to use "start_time". But the build number is safe enough.
+# The time between queue and start is less than 10 seconds whereas the time to setup the env is 20 second.
+# If this case happen, both build will be aware of the existence of the other one.
 def other_job_is_running?(uri, job, build_num)
   response = Net::HTTP.get(uri)
   builds = JSON.parse(response)
 
   builds.any? do |build|
-    build.dig("workflows", "job_name").to_s == job && build["build_num"] != build_num
+    if build.dig("workflows", "job_name").to_s == job && build["build_num"].to_i < build_num
```

Comment:
**question**

Not sure if that's important or the possible consequences but isn't it possible to have "build number M" < "build number N" while having more recent code in build M ?

Reply:
Thinking about it, I feel like the only consequences would be that we may deploy a code newer than another one which may not be that big of a deal ? The older code should be prevented to be deployed later on thanks to `https://github.com/jobteaser/circleci/blob/master/orbs/helm/deployable`.
I think it should be safe... Your thoughts ?